File: irlcpc/2024/1_CiaraCountlessCalculationCorrections.cpp
#include <iostream>
#include <string>

int main() {
    int N;
    std::cin >> N;
    int right = 0;
    for (int i = 0; i < N; i++) {
        int op1, op2, expectedResult;
        std::string equals;
        std::string operation;
        std::cin >> op1 >> operation >> op2 >> equals >> expectedResult;
        int actualResult;
        switch (operation[0]) {
            case '+':
                actualResult = op1 + op2;
                break;
            case '-':
                actualResult = op1 - op2;
                break;
            case '*':  
                actualResult = op1 * op2;
                break;
            case '/':
                actualResult = op1 / op2;
                break;
        }
        if (actualResult == expectedResult) {
            right++;
        }
    }
    int percentage = (right * 100) / N;
    char grade = 'F';
    if (percentage > 70) {
        grade = 'A';
    } else if (percentage > 60) {
        grade = 'B';
    } else if (percentage > 50) {
        grade = 'C';
    } else if (percentage > 40) {
        grade = 'D';
    } else if (percentage > 30) {
        grade = 'E';
    }
    std::cout << grade << std::endl;
    return 0;
}

File: irlcpc/2024/1_CiaraCountlessCalculationCorrections.py
# 1.0 s
# 16 MB

N = int(input())
right = 0
for i in range(N):
    op1, operator, op2, _, expected_result = input().split(" ")
    op1 = int(op1); op2 = int(op2); expected_result = int(expected_result)
    actual_result = 0
    match operator:
        case "+":
            actual_result = op1 + op2
        case "-":
            actual_result = op1 - op2
        case "*":
            actual_result = op1 * op2
        case "//":
            actual_result = op1 // op2
    if actual_result == expected_result:
        right += 1

grade = "F"
percentage = right / N
if percentage > 0.7:
    grade = "A"
elif percentage > 0.6:
    grade = "B"
elif percentage > 0.5:
    grade = "C"
elif percentage > 0.4:
    grade = "D"
elif percentage > 0.3:
    grade = "E"

print(grade)